botocove/cove_runner.py

```diff
@@ -33,19 +38,31 @@ def __init__(
         self.thread_workers = thread_workers
 
     def run_cove_function(self) -> CoveFunctionOutput:
-        # Run decorated func with all valid sessions
-        with ThreadPoolExecutor(max_workers=self.thread_workers) as executor:
-            jobs = [executor.submit(self.cove_thread, s) for s in self.sessions]
-            outputs = (f.result() for f in as_completed(jobs))
+        def submit_decorated_func_with_all_sessions(
+            executor: ThreadPoolExecutor,
+        ) -> FutureCoveResults:
+            return [executor.submit(self.cove_thread, s) for s in self.sessions]
+
+        def iterate_results_in_order_of_completion(
```

Comment:
I like this idea - much more digestable. Only suggestion is not having it inside closure of run_cove_function, no functional benefit - can be outside the class itself as a generic helper

Reply:
The idea of using an inner function was to make it clearer that it's just a helper for the enclosing function. It's impossible to reference it from outside.

But I think the generic helper is actually more readable, so I'll go with that.

As a bonus, VS Code will warn about unused functions whose names start with an underscore, so it achieves the same thing as what I was trying with the inner function.